src/Symbol/Loader/FileSymbolLoader.php

```diff
@@ -104,4 +121,27 @@ public function withBaseDir(?string $baseDir): SymbolLoaderInterface
 
         return $clone;
     }
+
+    /**
+     * @param array{
+     *      psr-0?: array<string, string|string[]>,
+     *      psr-4?: array<string, string|string[]>,
+     *      classmap?: list<string>,
+     *      files?: list<string>
+     * } $autoloadDefinition
+     *
+     * @return array<string, list<string>>
+     */
+    private function normalizePsrStructure(array $autoloadDefinition): array
+    {
+        $callback = static function ($value): array {
+            if (is_array($value)) {
+                return $value;
+            }
+
+            return [$value];
+        };
+
+        return array_map($callback, $autoloadDefinition);
```

Comment:
Since we do have php7.4, just use an inline arrow function :)

Reply:
Give me some time. Need to figure out something personal. Will come back to you asap.

Reply:
no rush mate, take your time. I'll try to figure out from my side too

Reply:
Just tried to replicate it. I think the array shape for `normalizePsrStructure` is wrong.
It should accept `array<int|string, string|string[]>` which is a combination of all autoloadTypes (psr4, psr0, files and classmap)

And it should then return `array<int|string, list<string>>`.

The current error is, because every key from the current array shape returns an array. THis is why phpstan is arguing about the "always true" condition for the ternary operator.

```
    /**
     * @param array<int|string, string|string[]> $autoloadDefinition
     *
     * @return array<int|string, list<string>>
     */
    private function normalizePsrStructure(array $autoloadDefinition): array
    {
        return array_map(static fn ($value): array => is_array($value) ? $value : [$value], $autoloadDefinition);
    }
*/

Then you can update from there.

Reply:
> I think the array shape for normalizePsrStructure is wrong.

Ah I see, absolutely spot on. I corrected the signature, and inlined the arrow function. thanks for pointing it out.